### DISCUSSION TITLE
Git Clone is really slow on my computer, and sometimes doesn't work.

### DISCUSSION THREAD

**Discussion by ghost** [community/community#142264] [Programming Help] [answered]
### Body

Hi! I'm trying to get help for the `git clone` command, as lately I've been encountering some issues with it. I clone using SSH, as I prefer it over HTTP, and don't really want to switch to HTTP.

## The Issue

Whenever I'm cloning a GitHub Repo that is of medium-large size (For example the GitHub Docs repo), it will always get stuck on some random process, for example, `Receiving objects:   0% (1/703411)`, is a process it can get stuck on, and it will just be stuck there forever.  The only time cloning with SSH works is when I'm cloning a repository with minimal files, such as a empty repository, or a repository with just the README.md file.

Could someone please give me a solution to this? Thank you so much!

## Additional Context

A first I thought it was my internet but the download speed is quite good, so I didn't really think that was the issue. My only other theory is that my SSH Key has a few errors in it, and that I should generate a new one. 

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by JohnWilliston** [comment, marked as answer]
Git just sucks. And GitHub is a Microsoft-owned product, so you can imagine how reliable and performant that is. If you don't need complete history in your repo, you can add the `--depth 1` parameter to your clone command to do a "shallow clone" to cut down on the network traffic. I have a solid 1 Gbps fiber pipe, and GitHub fails or stalls at least 15% of the time.

**Reply by ghost** [reply]
Thanks for explaining, I guess there's not really much I can do
